docs-ref-services/key-vault.md

```diff
@@ -1,138 +1,107 @@
+
+To find the object ID for the service principal, go to the [Azure portal](https://portal.azure.com), navigate to your subscription, select **Access Control (IAM)**, select **Role assignments**, and locate the appropriate service principal. Select that service principal to open it, then find the object ID on the **Overview** page.
 
 ```python
-from azure.keyvault import KeyVaultClient
+# Authentication with a service principal
 from azure.common.credentials import ServicePrincipalCredentials
 
-credentials = ServicePrincipalCredentials(
-    client_id = '...',
-    secret = '...',
-    tenant = '...'
-)
-
-client = KeyVaultClient(credentials)
-
-# VAULT_URL must be in the format 'https://<vaultname>.vault.azure.net'
-# SECRET_VERSION is required, and can be obtained with the KeyVaultClient.get_secret_versions(self, vault_url, secret_id) API
-secret_bundle = client.get_secret(VAULT_URL, SECRET_ID, SECRET_VERSION)
-secret = secret_bundle.value
-```
-
-> [!div class="nextstepaction"]
-> [Explore the Client APIs](/python/api/overview/azure/keyvault/client)
-
-### Management library
+# Replace the values with your specific IDs
+client_id = "your_client_id"
+tenant_id = "your_tenant_id"
+app_secret = "your_app_secret"
 
-```bash
-pip install azure-mgmt-keyvault
+credentials = ServicePrincipalCredentials(client_id=client_id, secret=app_secret, tenant=tenant_id)
 ```
 
-### Example
-
-The following example shows how to create an Azure Key Vault. 
+With the `credentials` variable initialized, you can then use the following code to create an Azure Key Vault named `msdocs-vault01` in an *existing* resource group named `KeyVault-rg`. With the Key Vault in place, you can then use the [client library](#client-library) to create, retrieve, and delete items.
 
 ```python
+# You can place this import statement earlier in your code.
 from azure.mgmt.keyvault import KeyVaultManagementClient
-from azure.common.credentials import ServicePrincipalCredentials
 
+# The Azure subscription to use. NOTE: hard-coding such an ID is not recommended and is
+# shown here for demonstration purposes only. In production code, use an environment
+# variable or other means to store and retrieve the ID.
+subscription_id = 'your_subscription_id'
 
-credentials = ServicePrincipalCredentials(
-    client_id = '...',
-    secret = '...',
-    tenant = '...'
-)
-
-# Even when using service principal credentials, a subscription ID is required. For service principals,
-# this should be the subscription used to create the service principal. Storing a token like a valid
-# subscription ID in code is not recommended and only shown here for example purposes.
-SUBSCRIPTION_ID = '...'
-client = KeyVaultManagementClient(credentials, SUBSCRIPTION_ID)
+client = KeyVaultManagementClient(credentials, subscription_id)
```

Comment:
... which is to say, this client will eventually raise an error if `credentials` is `DefaultAzureCredential`.